addons/web_editor/static/src/components/media_dialog/video_selector.js

```diff
@@ -104,7 +104,7 @@ export class VideoSelector extends Component {
             if (this.props.media) {
                 const src = this.props.media.dataset.oeExpression || this.props.media.dataset.src || (this.props.media.tagName === 'IFRAME' && this.props.media.getAttribute('src')) || '';
                 if (src) {
-                    this.state.urlInput = src;
+                    this.state.urlInput = "https:"+src;
```

Comment:
- Directly prepending "https" to the src without checking whether it already includes a protocol (e.g., http://, https://, etc.) can lead to malformed URLs. For example, if src is "http://example.com", this change will result in "httpshttp://example.com", which is not valid.
Introduce a check to ensure that the src does not already include a protocol. You can use URL or a regex to verify this.
- Use template literals `${}` for better readability and maintainability. This is a preferred practice in modern JS.

Maybe something like that? :

```suggestion
                    const urlWithProtocol = src.match(/^https?:\/\//) ? src : `https://${src}`;
                    this.state.urlInput = urlWithProtocol;
```

Reply:
@SergeBayet Even if you have `http` or `https` in your URL, embed URL returned by RPC made in `_getVideoURLData()` method will not contain the protocol part. Reason is that `get_video_source_data()` in tools.py which returns formatted URL doesn't contain protocol.

Reply:
Indeed, but if we change the behavior of `get_video_url_data()` in the `tools.py` for any reason to return absolute urls, it will be broken. But I let you decide what is best.

Reply:
@SergeBayet The solution looks good to me, as it is better to have changes in one place rather than in multiple places, as both approaches work the same way.
Thanks.. 😃